### PR TITLE
Skeleton of the Model Plugin Deployer v2 in Rust

### PR DESCRIPTION
**Pull request by wimax-grapl**
### Which issue does this PR correspond to?
https://app.zenhub.com/workspaces/grapl-6036cbd36bacff000ef314f2/issues/grapl-security/issue-tracker/436
https://github.com/grapl-security/issue-tracker/issues/436

### What changes does this PR make to Grapl? Why?
This is actually somewhat ancient (pre-Nomad) work, ported to work on Nomad.
It's a barebones functional grpc service (with integration tests!) - no actual MPD functionality in there yet besides a protobuf.

### How were these changes tested?
Integration tests!!!

### CODE REVIEW

File: src/proto/graplinc/grapl/api/model_plugin_deployer/v1/model_plugin_deployer_health.proto [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Inspired by https://github.com/grpc/grpc-proto/blob/master/grpc/health/v1/health.proto
+
+syntax = "proto3";
+
+package grpc.health.v1;
+
+option csharp_namespace = "Grpc.Health.V1";
```
Comment (jgrillo-grapl) [inline src/proto/graplinc/grapl/api/model_plugin_deployer/v1/model_plugin_deployer_health.proto:7]: we probably don't need any of these language-specific declarations

File: src/proto/graplinc/grapl/api/model_plugin_deployer/v1/model_plugin_deployer_health.proto [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// Inspired by https://github.com/grpc/grpc-proto/blob/master/grpc/health/v1/health.proto
+
+syntax = "proto3";
+
+package grpc.health.v1;
```
Comment (jgrillo-grapl) [inline src/proto/graplinc/grapl/api/model_plugin_deployer/v1/model_plugin_deployer_health.proto:5]: should this be a `grapl` package?

File: src/rust/model-plugin-deployer/src/server.rs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    metrics::register_counter!("request_count", "count of requests made to endpoint");
+    metrics::register_histogram!("request_ns", "nanoseconds for request execution");
+
+    tracing::info!(
+        message="About to start ModelPluginDeployer + HealthServer",
+        addr=?socket_addr,
+    );
+
+    Server::builder()
+        //.add_service(health_service)
```
Comment (jgrillo-grapl) [inline src/rust/model-plugin-deployer/src/server.rs:84]: Does this need to be commented out? If so, why?

File: src/rust/model-plugin-deployer/tests/test_deploy_model.rs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+use model_plugin_deployer::client::{
+    DeployModelRequest,
+    ModelPluginDeployerRpcClient,
+};
+
+use tonic::{
+    Code,
+};
+
```
Comment (jgrillo-grapl) [inline src/rust/model-plugin-deployer/tests/test_deploy_model.rs:12]: maybe exercise the healthcheck endpoints here too?

**Review by jgrillo-grapl — APPROVED** [review]
LGTM overall, lints are failing, and a few comments/questions

File: src/rust/model-plugin-deployer/benches/model_plugin_deployer_benchmark.rs [hunk 5]
```diff
@@ -0,0 +1,13 @@
+use criterion::{
```
Comment (christophermaier) [inline src/rust/model-plugin-deployer/benches/model_plugin_deployer_benchmark.rs:1]: Is this wired up to run in CI?

File: src/rust/model-plugin-deployer/tests/test_deploy_model.rs (hunk 4 above)
Comment (wimax-grapl) [reply src/rust/model-plugin-deployer/tests/test_deploy_model.rs:12]: A) Good idea
B) I want to shift over to other more important projects - this is more about getting this project out of PR hell - so I filed a ticket: https://github.com/grapl-security/issue-tracker/issues/760

File: src/rust/model-plugin-deployer/tests/test_deploy_model.rs [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+#[tokio::test]
+async fn test_deploy_model() -> Result<(), Box<dyn std::error::Error>> {
+    let mut client = ModelPluginDeployerRpcClient::from_env().await?;
```
[the remark this answers is not in the document]
Comment (christophermaier) [reply src/rust/model-plugin-deployer/tests/test_deploy_model.rs]: Docstrings at either the top of the module or attached to the `test_host`/`test_port` helper functions could also explain that the server is running in Nomad somewhere.

**Review by christophermaier — APPROVED** [review]

**Pull request merged**